composer.json

```
@@ -28,22 +31,27 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "php": "^7.4|^8.0"
+        "php": "^8.0",
+        "psr/container": "^1.0|^2.0",
+        "yiisoft/definitions": "^1.0",
+        "yiisoft/factory": "^1.0",
+        "yiisoft/validator": "dev-refactoring-v2"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.5",
         "roave/infection-static-analysis-plugin": "^1.16",
         "spatie/phpunit-watcher": "^1.23",
-        "vimeo/psalm": "^4.18"
+        "vimeo/psalm": "^4.18",
+        "yiisoft/test-support": "^1.4"
     },
     "autoload": {
         "psr-4": {
-            "Yiisoft\\_____\\": "src"
+            "Yiisoft\\Validator\\Rules\\Container\\": "src"
```

```suggestion
            "Yiisoft\\ValidatorRuleContainer\\": "src"
```
Suggestion only. I'm not sure it's better this way.

---

I think `Yiisoft\Validator\Rule\Handler\Container` is ok.